File: python/algorithms/selection.py
from util import swap_int

def selection_sort(arr):
    n = len(arr)
    if n < 2:
        return
    for i in range(n - 1):
        min_index = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j
        if min_index != i:
            swap_int(arr, i, min_index)

File: c/algorithms/selection.c
#include "selection.h"
#include "../util.h"  

void selection_sort(int *arr, size_t n) {
    if (!arr || n < 2) return;

    for (size_t i = 0; i < n - 1; i++) {
        size_t min_index = i;
        for (size_t j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_index]) {
                min_index = j;
            }
        }
        if (min_index != i) {
            swap_int(&arr[i], &arr[min_index]);
        }
    }
}
